### PR TITLE
Feature/sightings

### PR DESCRIPTION
**Pull request by jasmyn2244**
____ Wrote Tests __x__ Implemented ____ Reviewed

## Neccesary checkmarks:

    [x] All Tests are Passing in all environments

    [] The code will run locally

## Type of change

    [x] New feature
    [] Bug Fix
    [] Refactor

## Description of Change:

    description: 
- Added Single Cryptid Sighting Component
- Added filter by cryptid functionality to All Sightings
   

## Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

## Testing Changes

    [x] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

    [x] My code has no unused/commented out code
    [x] My code has no binding.pry's
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

### CODE REVIEW

File: cypress/integration/singleCryptid-test.js [hunk 1]
```diff
@@ -3,18 +3,18 @@ describe('All sightings page cypress test', () => {
...
-  
+
   it('Should confirm that true is equal to true', () => {
     expect(true).to.equal(true)
     expect(true).to.equal(true)
   });
 
   it('Should start on the home page and click on a cryptid', () => {
-    cy.get('\#\\32').click()
+    cy.get('\#\\31 0').click()
```
Comment (RosaTheDev) [inline cypress/integration/singleCryptid-test.js:13]: Nice job with fixing these test Jasmyn! Straight up Jimmy Nutron Style!

File: src/Components/CryptidCard.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 const CryptidCard = ({ id, name, image }) => {
+  const { setCryptid } = useContext(CryptidContext)
+
   return (
     <>
-    <Link to={`/singlecryptid/${name}`} className='card' id={id}>
-      <div className='name'>{name}</div>
-      <img src={image} alt='bigfoot' className='card-img' />
-    </Link>
+      <Link to={`/singlecryptid/${name}`} className='card' id={id} onClick={() => setCryptid(name)}>
```
Comment (RosaTheDev) [inline src/Components/CryptidCard.js:11]: Nice job with the on click functionality

File: src/App.js [hunk 3]
```diff
@@ -9,19 +9,24 @@ import NavBar from './Components/NavBar';
...
-        </Routes>
+        <CryptidProvider>
+          <NavBar />
+          <ErrorModal />
+          <Routes>
+            <Route path="/" element={<Home />} />
+            <Route path="/singlecryptid/:name" element={<SingleCryptid />} />
+            <Route path="/allsightings" element={<AllSightings />} />
+            <Route path="/singlesighting/:name" element={<SingleSighting />} />
+            <Route path="/singlecryptidsightings/:name" element={<SingleCryptidSightings />} />
```
Comment (krdurrett) [inline src/App.js:27]: Smart to have a new component for single cryptids

**Review by RosaTheDev — APPROVED** [review]

File: src/Components/AllSightings.js [hunk 4]
```diff
@@ -1,20 +1,89 @@
...
+        location
+        image
+      }
+    }
+  }
+`
 
 const AllSightings = () => {
-  const [ display, setDisplay ] = useState(false)
+  const { data, error, loading } = useQuery(GET_ALL_SIGHTINGS)
```
Comment (krdurrett) [inline src/Components/AllSightings.js:22]: Smart to set pageData for logic to show all sightings or cryptid specific sighting

File: src/App.js (hunk 3 above)
Comment (GraceGardner) [reply src/App.js:27]: Great idea to be able to use the link for routing.

File: src/Components/AllSightings.js [hunk 5]
```diff
@@ -1,20 +1,89 @@
...
+    console.log("pageData in resetData: ", pageData)
+  }
+
+  if (loading) return "Loading..."
+
+  if (!loading && !pageData) return setPageData(data.getCryptids)
+
+  if (error) return setError(error)
+
+  // console.log("pageData: ", pageData)
```
Comment (krdurrett) [inline src/Components/AllSightings.js:61]: Don't forget to get rid of console.logs

File: src/Context/CryptidContext.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React, { useState } from 'react'
```
Comment (krdurrett) [inline src/Context/CryptidContext.js:1]: Using cryptid context is such a smart idea

File: src/Components/AllSightings.js [hunk 7]
```diff
@@ -1,20 +1,89 @@
 import '../Styles/AllSightings.scss';
 import SightingCard from './SightingCard';
-import React, {useState} from 'react';
```
Comment (GraceGardner) [inline src/Components/AllSightings.js:3]: Good catch getting rid of these

**Review by krdurrett — APPROVED** [review]
So many great changes! Don't forget to delete console logs

File: src/Components/AllSightings.js [hunk 8]
```diff
@@ -1,20 +1,89 @@
...
-  const [ display, setDisplay ] = useState(false)
+  const { data, error, loading } = useQuery(GET_ALL_SIGHTINGS)
+  const [display, setDisplay] = useState(false)
+  const [pageData, setPageData] = useState(null)
+  const { setError } = useContext(ErrorContext)
+  const { setClick } = useContext(NavigationContext)
+
+  useEffect(() => {
+    setClick(true)
+    //setPageData(data)
```
Comment (GraceGardner) [inline src/Components/AllSightings.js:30]: We can delete this now that it's in a better home.

File: src/Components/AllSightings.js (hunk 5 above)
Comment (GraceGardner) [inline src/Components/AllSightings.js:61]: could remove these console.logs

File: src/Components/SingleCryptidSightings.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            <button
+              className='cryptid-button'
+              onClick={toggleDisplay}
+            >
+              Search by Cryptids
+            </button>
+            {display &&
+              <div className='dropdown-container'>
+                <p>things go here</p>
+              </div>
```
Comment (GraceGardner) [inline src/Components/SingleCryptidSightings.js:72]: Do we need this stuff that's commented out?

**Review by GraceGardner — APPROVED** [review]

**Pull request merged**